### DISCUSSION TITLE
Why the product images appear only when you hover the mouse on them?

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2209] [Ideas]
Go to Backend > Catalog > Manage Products. Choose a product and edit it. No images are displayed in the Images tab initially. In order to be displayed. You have to hover the mouse pointer over each one.

This behavior gets annoying when there are many images and you have to edit their labels, positions or choose which one is Base Image/Small Image/Thumbnail. 

I haven't searched the Internet so far but I would be interested in why the Magento team chose not to display the images from the very beginning. Is there a way for these images to appear without to hover the mouse on each of them?

**Comment by fballiano** [comment]
I think that's just to avoid wasting bandwidth, since those images are uncompressed (original size) and could be megabytes big.

**Reply by luigifab** [reply]
You can create thumbnails with resize method.

Many years ago I replaced the template with thumbnails and zoom:
![image](https://user-images.githubusercontent.com/31816829/173047116-88e4456e-81fb-4777-9766-8ff1ce4ebce9.png)

**Reply by addison74** [reply]
I didn't think of that but now it's obvious. The template must be modified and resized images (width 100px should be fine) used to solve both issues, downloading images to their original size and viewing them only with the mouse over.

This is the code now for that cell inside the grid:

```html
<td class="cell-image"><div class="place-holder" onmouseover="media_gallery_contentJsObject.loadImage('/a/d/adidas-m-test-1.jpg')" style="display: none;"><span>Roll Over for preview</span></div><img src="https://www.mydomain.tld/media/catalog/product/a/d/adidas-m-test-1.jpg" width="100" style="" alt=""></td>
```

They chose to use the original image (the largest one) and see it only when hovering the mouse pointer over it. I have never seen this approach before. When you have 20 images and you want to change their positions, label, exclude, you have to use the mouse and for every image the browser makes a request for the original image, which it is resized with width property to 100px. This approach left me speechless.  I think it would be an improvement for OM as well. Can you post your template?

**Reply by luigifab** [reply]
It is [here](https://github.com/luigifab/openmage-apijs/blob/master/src/app/design/adminhtml/default/default/template/luigifab/apijs/gallery.phtml), but I think you can't use it as is (without the module).

**Reply by fballiano** [reply]
I don't know, I've a customer with hundreds of thousand of images and thumbnails where always a problem (they tend to explode in number, they're multiplied for every store, we've thumbnail, small_image, image), and they prefer to have the full size image there for their own quality checks so... adding a new resized image thumbnail... i don't know...

**Reply by addison74** [reply]
@luigifab - at least I am inspired to evaluate and make the change. I only want the images to appear without mouse actions and be 100px in size, resized from the original.

**Reply by addison74** [reply]
@fballiano - In my case the master images are obtained by automatic processing in Photoshop. From here they are saved to the desired size and quality then uploaded. It is no longer necessary to enlarge the ones displayed in the grid for quality checking but only to see them as thumbnails and perform actions on them (like setting position, label, exclude, remove) without wasting time with the mouse. 

The moment the mouse hovers an image and reveal it the browser makes a request to the server and it is displayed resized with width, so the bandwidth is not saved at all. When the filled in label is visible then you don't need to see the images. 

Mouse movement becomes routine but when you waste time with this action and it becomes stressful it is a problem. My opinion is this feature was designs for ancient time when Magento was released and there were no high Internet speeds like today.

**Reply by elidrissidev** [reply]
Lazy loading may be an option too, they'll only be fetched by the browser when you click the Images tab

**Reply by addison74** [reply]
I looked at both the Magento implementation and @luigifab implementation. To replace the line bellow in the template and use the resize method for product images some pretty interesting changes are needed.

```php
<td class="cell-image"><div class="place-holder" onmouseover="<?php echo $_block->getJsObjectName(); ?>.loadImage('__file__')"><span><?php echo Mage::helper('catalog')->__('Roll Over for preview') ?></span></div><img src="<?php echo $this->getSkinUrl('images/spacer.gif')?>" width="100" style="display:none;" alt="" /></td>
```

I did not approach lazy loading.

**Comment by addison74** [comment]
The Magento team's concern about not loading product images from the very beginning does not cover all scenarios. If I edit a product, upload its images then I save it, if in the same session I return to edit the same product or just press [Save and Continue Edit] button, ]the image is already in the browser cache (if not it is not disabled), but the images need to roll over again which is no longer needed.
